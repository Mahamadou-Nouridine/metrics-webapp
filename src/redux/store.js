import { configureStore } from '@reduxjs/toolkit';
import { reducer } from './country/coutriesSLice';

export default configureStore({
  reducer: {
    countries: reducer,
  },
});
