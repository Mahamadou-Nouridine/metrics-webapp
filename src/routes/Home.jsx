import React from 'react';
import Headline from '../Components/Headline';
import Search from '../Components/Search';
import CountriesList from '../Components/CountriesList';

const Home = () => (
  <>
    <Headline />
    <Search />
    <CountriesList />
  </>
);

export default Home;
