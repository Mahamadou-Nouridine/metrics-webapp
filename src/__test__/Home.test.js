import React from 'react';
import { Provider, useSelector } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import Home from '../routes/Home';
import store from '../redux/store';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('<Home />', () => {
  const initialState = {
    countries: [
      { name: 'Country 1', flag: 'https://country-1-flag.png', area: 30000 },
      { name: 'Country 2', flag: 'https://country-2-flag.png', area: 30000 },
      { name: 'Country 3', flag: 'https://country-3-flag.png', area: 30000 },
    ],
    error: undefined,
    isLoading: false,
    currentPage: '',
  };
  const mockedData = initialState;

  beforeEach(() => {
    useSelector.mockReturnValue(mockedData);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should displays all countries on load', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    const countryImages = screen.getAllByRole('img', { name: '' });
    expect(countryImages).toHaveLength(mockedData.countries.length);
  });

  it('Should filters countries based on search', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );

    const searchInput = screen.getByPlaceholderText('country name ...');
    fireEvent.change(searchInput, { target: { value: 'country 2' } });

    const filteredCountries = screen.getAllByRole('link', { name: /^country/i });
    expect(filteredCountries).toHaveLength(1);
  });
});
