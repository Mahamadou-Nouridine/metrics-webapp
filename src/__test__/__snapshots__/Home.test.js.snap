// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      World countries metrics
    </h1>
  </div>
  <div
    class="container"
  >
    <form>
      <h3>
        Search by country name
      </h3>
      <div
        class="inputContainer"
      >
        <input
          placeholder="country name ..."
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
  <div
    class="container"
  >
    <a
      class="container"
      href="/countries/Country 1"
    >
      <img
        alt=""
        src="https://country-1-flag.png"
      />
      <h3>
        Country 1
      </h3>
      <h3>
        30000
        km²
      </h3>
    </a>
    <a
      class="container"
      href="/countries/Country 2"
    >
      <img
        alt=""
        src="https://country-2-flag.png"
      />
      <h3>
        Country 2
      </h3>
      <h3>
        30000
        km²
      </h3>
    </a>
    <a
      class="container"
      href="/countries/Country 3"
    >
      <img
        alt=""
        src="https://country-3-flag.png"
      />
      <h3>
        Country 3
      </h3>
      <h3>
        30000
        km²
      </h3>
    </a>
  </div>
</div>
`;
